src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -234,7 +185,7 @@ private function createAutowiredDefinition(\ReflectionClass $typeHint, $id)
         $argumentDefinition = $this->container->register($argumentId, $typeHint->name);
         $argumentDefinition->setPublic(false);
 
-        $this->populateAvailableType($argumentId, $argumentDefinition);
+        $this->populateAvailableTypes();
         $this->completeDefinition($argumentId, $argumentDefinition);
```

Comment:
Is the `populateAvailableTypes()` call needed? That's *called* in `completeDefinition`. The fact that we have to call that manually, has some code smell. I think it might be better if there were a `getAutowiringTypes()` method, and this checks if `autowiringTypes` is null, and initializes it (once) if it is. Then, `populateAvailableType` could *return* the array of types, and `populateAvailableTypes` actually sets that on the `autowiringTypes` property. This is all minor, internal details - but it might make the code easier to follow.